src/portable/synopsys/dwc2/hcd_dwc2.c

```diff
@@ -456,11 +457,23 @@ tusb_speed_t hcd_port_speed_get(uint8_t rhport) {
 
 // HCD closes all opened endpoints belong to this device
 void hcd_device_close(uint8_t rhport, uint8_t dev_addr) {
-  (void) rhport;
+  dwc2_regs_t* dwc2 = DWC2_REG(rhport);
   for (uint8_t i = 0; i < (uint8_t) CFG_TUH_DWC2_ENDPOINT_MAX; i++) {
     hcd_endpoint_t* edpt = &_hcd_data.edpt[i];
     if (edpt->hcchar_bm.enable && edpt->hcchar_bm.dev_addr == dev_addr) {
       tu_memclr(edpt, sizeof(hcd_endpoint_t));
+      for (uint8_t ch_id = 0; ch_id < (uint8_t) DWC2_CHANNEL_COUNT_MAX; ch_id++) {
+        hcd_xfer_t* xfer = &_hcd_data.xfer[ch_id];
+        if (xfer->allocated && xfer->ep_id == i) {
+          dwc2_channel_t* channel = &dwc2->channel[ch_id];
+          dwc2_channel_split_t hcsplt = {.value = channel->hcsplt};
+          xfer->closing = 1;
+          // Channel disable must not be programmed for non-split periodic channels
```

Comment:
this is only true for DMA mode, for slave mode, we still need to disable it I guess 

<img width="982" height="192" alt="Image" src="https://github.com/user-attachments/assets/167b4f64-7f23-4075-989a-458ec7ae0e6c" />